### PR TITLE
update installation instructions to use astropy conda channel

### LINKED ISSUE #1451: Update installation guide for the 0.6 release
Specifically relating to the installation of wcsaxes from the astropy binstar repo.

### PR DESCRIPTION
**Pull request by Cadair**
closes #1451

### CODE REVIEW

**Comment by Cadair** [comment]
@ayshih I have just updated the hashes in this, so let's see if this works.

**Comment by ayshih** [comment]
Per [my other comment](https://github.com/sunpy/sunpy/pull/1432#issuecomment-117784845), I'm not sure I'm comfortable with putting in the `wcsaxes`-based hashes when `wcsaxes` is not yet a dependency.

**Comment by Cadair** [comment]
@ayshih it is the default if it is installed, so I would argue that the CI servers and the tests should use it by default.

It would also make it harder to test the other way around, because when testing locally if you did have it installed then it would break all the tests.

What does need to happen is we need some figure tests, to test the behaviour when it is not available.

**Comment by Cadair** [comment]
The travis fail here is fixed by #1479

File: doc/source/guide/installation/index.rst [hunk 1]
```diff
@@ -38,7 +38,7 @@ To install SunPy launch a system command prompt or the
 
 To configure conda for sunpy downloads run::
 
-    conda config --add channels sunpy
+    conda config --add channels sunpy --add channels astropy
```
Comment (dpshelio) [inline doc/source/guide/installation/index.rst]: I thought `astropy` was in conda by default...  why do we need to add their channel for?

Comment (Cadair) [inline doc/source/guide/installation/index.rst]: wcsaxes, https://binstar.org/astropy

**Comment by ayshih** [comment]
Ideally, we'd want the tests to be able to pass irrespective of whether `wcsaxes` is installed.  If that's not possible, at least we'd want the test failures to make clear to a user *why* the test failed.

Here are some ways to proceed that I can think of:
* The path of least work is to simply accept the non-informative test failures if a user doesn't have `wcsaxes`.
* We have the figure tests raise an ImportError if `wcsaxes` is not available instead of letting them reach the inevitable hash mismatch.  This check can be incorporated into the decorator wrapper so it's in one place, and then removed once `wcsaxes` is a dependency.
* We keep the non-`wcsaxes` hashes and have some sort of API or configuration toggle to inhibit the use of `wcsaxes` even if available.

**Comment by Cadair** [comment]
I can think of a couple of ways to deal with this.

1. We have the figure test decorator add to the error message that `wcsaxes` is not installed and my be the reason the tests are failing.
2. We have a new pytest mark which is for tests that require wcsaxes, and then we can automatically deselect the figure tests that require wcsaxes when wcsaxes is not present.

This would mean skipping the current figure tests if wcsaxes is not installed, but then we could write new tests that test the non-wcsaxes versions, that would run irrespective of having wcsaxes installed.

**Comment by dpshelio** [comment]
@Cadair I like your second point, which it will make the whole thing more complete... but that would require more work.  If we are going to have `wcsaxes` as a dependency... then... that extra work would be just for this version, right?

**Comment by Cadair** [comment]
@dpshelio yes, at the point we were to make wcsaxes a requirement it would be redundant. However, I do actually wonder if we will want to do that in the short term. wcsaxes is awesome, but preventing users from not using it, for special edge case plotting might not be the best idea.

p.s. WHY AM I GIVING MYSELF MORE WORK.

**Comment by wafels** [comment]
Let me get this straight - if wcsaxes is installed, then all map plotting
uses it, and you can't go back to *not* using it without uninstalling it?
Is that how the map software works in master?

**Comment by Cadair** [comment]
@wafels no you can not use it in a couple of different ways, the easiest one being pass a non-wcsaxes axes instance to plot.

i.e.
```python
fig, ax = plt.subplots()

mymap.plot(axes=ax)
```

would not use wcsaxes.

We are talking about the tests, which in this case are using the default behaviour (to use wcsaxes when installed) and therefore are breaking if you don't have wcsaxes installed.

**Comment by wafels** [comment]
I realize you are talking about the tests.  So your code snippet a way to
force the existing plot tests to always not use wcsaxes? My quibble is that
installation of wcsaxes changes the default of behaviour of map plotting.
Is that change in default behaviour desirable?

**Comment by Cadair** [comment]
@wafels my argument is that the wcsaxes behaviour is superior and therefore should be used by default. The PR that made that change has already been merged.

Yes, that snippet or a similar one is probably the best way to force a test to not to use wcsaxes, there is also a way to turn of wcsaxes globally by changing a constant in the visualisation module.

**Comment by ayshih** [comment]
My preference is to modify the figure-test decorator to explicitly set `visualization.FORCE_NO_WCSAXES` to True for the duration of the figure test, and keep the non-`wcsaxes` hashes.

**Comment by Cadair** [comment]
@ayshih seeing how most users will be using it with wcsaxes (it will be installed by conda by default) this seems counter-intuitive. Also, if we go that route, it will be impossible to write tests with wcsaxes on.

**Comment by Cadair** [comment]
The behaviour we have with other optional libs, like glymur at the moment, is that if it is not installed tests are just skipped, which is what I was suggesting with my # 2 suggestion above.

**Comment by ayshih** [comment]
@Cadair This solution is just to get us a 0.6 release that can have figure tests that run and pass irrespective of whether `wcsaxes` is available.  The figure tests are still useful even if `wcsaxes` is forced off; it just doesn't test `wcsaxes`.

Wasn't the intention to make `wcsaxes` a dependency with 0.7?  Once 0.6 is out, we'd remove the forced off, and then update all of the hashes.

**Comment by Cadair** [comment]
@ayshih if you are worried about time, doing the skip if no wcsaxes thing will take about 10 mins. Adding a couple of non-wcsaxes tests, another 10, it's not really a big deal.

Or are you more philosophically objecting to it?

**Comment by ayshih** [comment]
Well, my objection is partially philosophical, but I also figured that my solution would take even less time to implement.  But I don't currently have time to do it, so if you want to implement your solution, go ahead. =)

**Comment by Cadair** [comment]
@ayshih I think this solution is the best of both worlds, we are testing plots with both wcsaxes and not wcsaxes if no wcsaxes is installed, and only non-wcsaxes plots if it's not installed.

**Comment by ehsteve** [comment]
I am for whatever brings the release of 0.6 closer to the present!

**Comment by Cadair** [comment]
Well it's done...

**Comment by ayshih** [comment]
It looks good, but I find our naming of these decorators confusing, e.g.:

* `skip_windows`: skip if the platform is Windows
* `skip_glymur`, `skip_ana`, `skip_wcsaxes`: skip if the package is not available

There is an inconsistency of sense on whether whether the test is skipped on availability or lack of availability.  I'd also prefer a naming scheme that is less a directive and more a descriptor for the test.  How about this naming scheme, e.g.:

* `requires_non_windows`
* `requires_glymur`, `requires_ana`, `requires_wcsaxes`

**Comment by ayshih** [comment]
:+1: (my naming-scheme concern is not a blocking issue for this PR)

**Comment by ehsteve** [comment]
:+1: this is clearly a band-aid on a larger problem but we need to move forward.

**Pull request merged**